### PR TITLE
[controller] Minor code cleanup

### PR DESCRIPTION
**Pull request by lluwm**
This rb fixes several problems found by our IDE framework without modifying any semantics.

VeniceControllerWrapper.java
-- remove unnecessary initialization to empty string.
-- remove unnecessary null check before 'instanceof'.
-- remove unused functions.

VeniceController.java
-- stop controller services in the reverse order as they are started.

StoreBackupVersionCleanupService.java
StoreGraveyardCleanupService.java
-- use AtomicBoolean to avoid potential race.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceControllerWrapper.java [hunk 1]
```diff
@@ -232,12 +231,13 @@ static StatefulServiceProvider<VeniceControllerWrapper> generateService(VeniceCo
...
 
-        String fabricAllowList = "";
+        String fabricAllowList;
         if (options.isParent()) {
           // Parent controller needs config to route per-cluster requests such as job status
           // This dummy parent controller won't support such requests until we make this config configurable.
           // go/inclusivecode deferred(Reference will be removed when clients have migrated)
-          fabricAllowList = extraProps.getStringWithAlternative(CHILD_CLUSTER_ALLOWLIST, CHILD_CLUSTER_WHITELIST, "");
+          fabricAllowList =
+              extraProps.getStringWithAlternative(CHILD_CLUSTER_ALLOWLIST, CHILD_CLUSTER_WHITELIST, StringUtils.EMPTY);
```
Comment (FelixGV) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceControllerWrapper.java:240]: Hmm... is it useful to use this constant? What's the difference?

**Review by FelixGV — APPROVED** [review]
LGTM, I just have a minor question.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceControllerWrapper.java (hunk 1 above)
Comment (lluwm) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceControllerWrapper.java:240]: Yes, they are the same. Just treat it as a predefined constant value with a bit more expressive name to me and refer to it. I don't mind reverting it back if `""` looks obvious enough.

Comment (FelixGV) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceControllerWrapper.java:240]: It's fine either way, as you wish.

**Pull request merged**